packages/sitecore-jss/src/layout-service.ts

```diff
@@ -201,3 +213,53 @@ export class RestLayoutService implements LayoutService {
     };
   }
 }
+
+export class GraphQLLayoutService implements LayoutService {
+  /**
+   * Fetch layout data using the Sitecore GraphQL endpoint.
+   * @param {GraphQLLayoutServiceConfig} serviceConfig
+   */
+  constructor(private serviceConfig: GraphQLLayoutServiceConfig) {}
+
+  /**
+   * Fetch layout data for an item.
+   * @param {string} itemPath
+   * @param {string} [language]
+   * @returns {Promise<LayoutServiceData>} layout service data
+   */
+  async fetchLayoutData(itemPath: string, language?: string): Promise<LayoutServiceData> {
+    const query = this.getLayoutQuery(itemPath, language);
+
+    const data = await this.createClient().request<{
+      layout: { item: { rendered: LayoutServiceData } };
+    }>(query);
+
+    return data.layout.item.rendered;
+  }
+
+  /**
+   * Returns new graphql client instance
+   */
+  private createClient(): GraphQLRequestClient {
+    const { endpoint } = this.serviceConfig;
+
+    return new GraphQLRequestClient(endpoint);
+  }
+
+  /**
+   * Returns GraphQL Layout query
+   * @param {string} itemPath page route
+   * @param {string} [language] language
+   */
+  private getLayoutQuery(itemPath: string, language?: string) {
```

Comment:
Would there be any reason one would want to override the layout query (similar to our sitemap service)? I can't think of a case for that, but just wanted to throw it out there in case there are scenarios you came across.

Reply:
I think Adam has a good point. Supporting multitenancy could be a scenario for this maybe?

Reply:
@anastasiya29 Is `siteName` not enough? I had a call with Anton, and he didn't provide me another query for multitenancy, so not sure when user can use it

Reply:
Added ability to customize it